common/datavalues/src/types/deserializations/nullable.rs

```diff
@@ -71,60 +71,84 @@ impl TypeDeserializer for NullableDeserializer {
 
     fn de_text_json<R: BufferRead>(
         &mut self,
-        reader: &mut R,
+        reader: &mut NestedCheckpointReader<R>,
         format: &FormatSettings,
     ) -> Result<()> {
+        reader.push_checkpoint();
```

Comment:
not find a good way to impl a guard for checkpoint


@sundy-li  

1.  most of the time, need to pop_checkpoint early like here, guard does not help.
2.  guard need to hold a mut ref to impl Drop,  so the flowing code need to use `guard.reader.read()`,  it is not convenient.  and can not pass it to inner consumer (like de_text_json here)

```
```

```
```